### PR TITLE
Fix sidebar height & hide ScrollUpButton until scrolled (fade in)

### LINKED ISSUE #383: Web ui: make sidebar full height
PR incoming:

UI looks like this when scrolling:

image

Fix it like so:

image

### PR DESCRIPTION
**Pull request by StewartThomson**
Fixes #383 and issue found within #276

### CODE REVIEW

File: ui/src/common/ScrollUpButton.tsx [hunk 1]
```diff
@@ -3,11 +3,36 @@ import KeyboardArrowUp from '@material-ui/icons/KeyboardArrowUp';
 import React, {Component} from 'react';
 
 class ScrollUpButton extends Component {
+    state = {
+        display: 'none',
+        opacity: 0,
+    };
+    componentDidMount() {
+        if (typeof window !== 'undefined') {
+            window.addEventListener('scroll', () => {
```
Comment (jmattheis) [inline ui/src/common/ScrollUpButton.tsx:12]: The handler probably should be removed on https://reactjs.org/docs/react-component.html#componentwillunmount

File: ui/src/common/ScrollUpButton.tsx [hunk 2]
```diff
@@ -3,11 +3,36 @@ import KeyboardArrowUp from '@material-ui/icons/KeyboardArrowUp';
...
         return (
             <Fab
                 color="primary"
-                style={{position: 'fixed', bottom: '30px', right: '30px', zIndex: 100000}}
+                style={{
+                    position: 'fixed',
+                    bottom: '30px',
+                    right: '30px',
+                    zIndex: 100000,
+                    display: `${this.state.display}`,
```
Comment (jmattheis) [inline ui/src/common/ScrollUpButton.tsx:33]:
```suggestion
                    display: this.state.display,
```

File: ui/src/common/ScrollUpButton.tsx [hunk 3]
```diff
@@ -3,11 +3,36 @@ import KeyboardArrowUp from '@material-ui/icons/KeyboardArrowUp';
...
+        display: 'none',
+        opacity: 0,
+    };
+    componentDidMount() {
+        if (typeof window !== 'undefined') {
+            window.addEventListener('scroll', () => {
+                let currentScrollPos = window.pageYOffset;
+                if (currentScrollPos > 0) {
+                    this.setState({display: 'inherit'});
+                    this.setState({opacity: currentScrollPos / 500});
```
Comment (jmattheis) [inline ui/src/common/ScrollUpButton.tsx:16]: Could you merge the 2 setStates into one? Same for the one below
```suggestion
                    this.setState({opacity: currentScrollPos / 500, opacity: Math.min(currentScrollPos / 500, 1)});
```
Also added Math.min that opacity doesn't exceed 1.

**Review by jmattheis — CHANGES_REQUESTED** [review]
Thanks for your contribution. This actually has been bugging me too.

I've added remarks in subcomments.

File: ui/src/common/ScrollUpButton.tsx (hunk 3 above)
Comment (jmattheis) [inline ui/src/common/ScrollUpButton.tsx:16]: Could you merge the 2 setStates into one? Same for the one below
```suggestion
                    this.setState({opacity: currentScrollPos / 500, opacity: Math.min(currentScrollPos / 500, 1)});
```
Also added Math.min that opacity doesn't exceed 1.

**Comment by StewartThomson** [comment]
Feedback implemented @jmattheis

File: ui/src/common/ScrollUpButton.tsx [hunk 4]
```diff
@@ -3,11 +3,43 @@ import KeyboardArrowUp from '@material-ui/icons/KeyboardArrowUp';
...
+    };
+    componentDidMount() {
+        if (typeof window !== 'undefined') {
+            window.addEventListener('scroll', this.scrollHandler.bind(this));
+        }
+    }
+
+    componentWillUnmount() {
+        if (typeof window !== 'undefined') {
+            window.removeEventListener('scroll', this.scrollHandler.bind(this));
```
Comment (jmattheis) [inline ui/src/common/ScrollUpButton.tsx:18]: I think this doesn't work as .bind will give a new reference and thus, doesn't unregister the old handler. I'll fix this. Also, I'll prevent some rerenders, by checking if the state will be changed.

**Review by jmattheis — APPROVED** [review]

**Pull request merged**